### PR TITLE
Add a scripts for conversion of all datasets

### PR DESCRIPTION
**Pull request by valosekj**
This PR adds the following scripts:

- `dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data.py`
- `dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data_pretraining.py` - you can specify `seg` or `lesion` to create dataset used either for pretraining or fine-tuning

TODO:
- [x] replace [convert_bids_to_nnUNetv2.py](https://github.com/ivadomed/model_seg_sci/blob/main/dataset-conversion/convert_bids_to_nnUNetv2.py) script by `dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data.py`?
- [x] add proper arg parser to `dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data.py`

### CODE REVIEW

**Review by naga-karthik — APPROVED** [review]
Since we tested the script and it is working, the PR is good to be merged once the remaining TODO is done!

**Comment by naga-karthik** [comment]
@valosekj can we merge with branch?

**Comment by naga-karthik** [comment]
Could you run the `convert_bids_to_nnUNetv2_all_sci_data.py` using only 1 dataset as the input (instead of the list of datasets)? If you fix seed=50 and check whether the test subjects are same for this dataset and the combined one that we used before, then we can be sure that `convert_bids_to_nnUNetv2_all_sci_data.py` script can also be used for individual datasets

**Comment by valosekj** [comment]
> Could you run the `convert_bids_to_nnUNetv2_all_sci_data.py` using only 1 dataset as the input (instead of the list of datasets)? If you fix seed=50 and check whether the test subjects are same for this dataset and the combined one that we used before, then we can be sure that `convert_bids_to_nnUNetv2_all_sci_data.py` script can also be used for individual datasets

I tested both the `convert_bids_to_nnUNetv2.py` and `convert_bids_to_nnUNetv2_all_sci_data.py` scripts with a fixed seed on `sci-zurich`:

`convert_bids_to_nnUNetv2_all_sci_data.py`:

```console
$ python ~/code/model_seg_sci/dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data.py --path-data sci-zurich --path-out ./ --dataset-name tSCIZurich --dataset-number 999 --seed 50 --split 0.8 0.2
2023-08-10 10:44:18.958 | INFO     | __main__:main:312 - ----- Dataset conversion finished! -----
2023-08-10 10:44:18.959 | INFO     | __main__:main:313 - Number of training and validation images (including sessions): 85
2023-08-10 10:44:18.959 | INFO     | __main__:main:314 - Number of test subjects (including sessions) in SCI-Zurich: 22
2023-08-10 10:44:18.959 | INFO     | __main__:main:315 - Number of test subjects (including sessions) in SCI-Colorado: 0
```

`convert_bids_to_nnUNetv2.py`:

```console
$ python ~/code/model_seg_sci/dataset-conversion/convert_bids_to_nnUNetv2.py  --path-data sci-zurich --path-out ./ --dataset-name tSCIZurich_convert_bids_to_nnUNetv2 --dataset-number 999 --seed 50 --split 0.8 0.2
2023-08-10 10:56:44.201 | INFO     | __main__:get_good_subjects:104 - Total number of subjects in the (curated) dataset: 97
2023-08-10 10:56:44.201 | INFO     | __main__:main:168 - Creating a dataset with 77 training subjects and 20 test subjects.
2023-08-10 10:56:55.745 | INFO     | __main__:main:309 - Number of training and validation subjects (including sessions): 85
2023-08-10 10:56:55.745 | INFO     | __main__:main:310 - Number of test subjects (including sessions): 22
```

The number of subjects in the output folders seems to be the same!

```console
$ ls -1 Dataset999_tSCIZurich/imagesTr | wc -l
85                
$ ls -1 Dataset999_tSCIZurich_convert_bids_to_nnUNetv2/imagesTr | wc -l
85  
$ ls -1 Dataset999_tSCIZurich/imagesTsZur | wc -l
22 
$ ls -1 Dataset999_tSCIZurich_convert_bids_to_nnUNetv2/imagesTs | wc -l
22
```

The only thing is that `convert_bids_to_nnUNetv2_all_sci_data.py` creates  `imagesTsZur` instead of `imagesTs`.

**Comment by naga-karthik** [comment]
Nice! Thanks for the comparison! I think we can add a condition to check `len(path_data)` and only create those additional folders if the length is `> 1`

**Comment by valosekj** [comment]
> Nice! Thanks for the comparison! I think we can add a condition to check `len(path_data)` and only create those additional folders if the length is `> 1`

Okay! Implemented in the recent commits.
